### ISSUE TITLE
Framework: CMake/CTest reporting problem

### ISSUE THREAD

**Issue by prwolfe** [trilinos/Trilinos#7731]
## Bug Report
@trilinos/framework 

### Description
As seen in https://github.com/trilinos/Trilinos/pull/7702#issuecomment-663574674 we are not getting all errors reported to the dashboard. This is due to a kitware bug detailed in https://gitlab.kitware.com/cmake/cmake/-/issues/19545.

This is here to track the issue on both ends.

**Comment by prwolfe** [comment]
@bartlettroscoe - just a heads up for your efforts with the kitware efforts.

**Comment by bartlettroscoe** [comment]
@prwolfe, can you summarize how critical/urgent this is for Trilinos?  I need to know to create and track this as part of the SNL Kitware contract work being managed in:

* https://gitlab.kitware.com/snl/project-1/-/boards

(I can get you access to that site so you can be part of the process to address this as well.)

**Comment by prwolfe** [comment]
@bartlettroscoe - I have access to that board looks like.  I am not sure that I should speak for Jim as to priority. I did ping Brad on his ticket to see what Kitware's priority is on this. I get the impression from reading the ticket that this has not yet been fully triaged.

The end result is that we get PR builds that show clean on CDash even though we get nice errors from ninja or make detailing the problem. This cost us about 4 hours on the Framework team in this last case and I'm not sure how long Chris spent frustrated about it. It happens 3-4 times per year, this is the first time we have traced it this far.

**Comment by bartlettroscoe** [comment]
@prwolfe, I too have seen cases where CDash did not report build errors.  I think it is very rare, but when it happens, it very confusing and frustrating.

I have made this a priority for our contract with Kitware.  See:

* https://gitlab.kitware.com/snl/project-1/-/issues/155

**Comment by prwolfe** [comment]
Thanks - it does frustrate all of us I'm sure. I was wondering if not using launchers would be a decent work-around and honestly I can't remember why we are using them. It's just been too long.

**Comment by bartlettroscoe** [comment]
> I can't remember why we are using them. It's just been too long.

@prwolfe, ironically (perhpas), you have to use the CMake compiler launchers in order to pick up and report the detailed build errors to CDash.  See:

* https://cmake.org/cmake/help/v3.13/prop_gbl/RULE_LAUNCH_COMPILE.html

@prwolfe, do we have a clear reproducer for Kitware to reproduce this behvaior so they can verify it is fixed (at least for our use case)?

which says:

> Makefile Generators and the Ninja generator prefix compiler commands with the given launcher command line. This is intended to allow launchers to intercept build problems with high granularity. Other generators ignore this property because their underlying build systems provide no hook to wrap individual commands with a launcher

**Comment by prwolfe** [comment]
I thought that might be the case ;->

A reproducer should be doable - I will have to spend a bit of time to get something that configures but doesn't compile and that's small. No reason to do a big project. It may be a week or so.

**Comment by bartlettroscoe** [comment]
CC: @zackgalbreath 

@prwolfe, FYI:
* https://gitlab.kitware.com/cmake/cmake/-/issues/19545
* https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5266

So I guess this will be fixed in CMake 3.19.0?

Our tax $$ hard at work!

**Comment by prwolfe** [comment]
I saw this in an e-mail. Very responsive and I'm looking forward to testing it.

Thanks!
